controllers/hbase_controller.go

```diff
@@ -491,9 +489,10 @@ func (r *HBaseReconciler) ensureStatefulSetPods(ctx context.Context, sts *appsv1
 
 	r.Log.Info("matched pods", "statefulset", sts.Name, "pods", len(podList.Items))
 
-	// sort pods by name to have things restarted predictably
+	// sort pods in reverse ordinal order to restart
+	//https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#rolling-update
 	sort.Slice(podList.Items, func(i, j int) bool {
-		return podList.Items[i].Name < podList.Items[j].Name
+		return podList.Items[i].Name > podList.Items[j].Name
```

Comment:
since it's comparing based on name, it still won't restart similar to how statefulsets are restarted. Because sorting would be like so for anything past single digit:

regionserver-9
...
regionserver-2
regionserver-11
regionserver-10
regionserver-1

Reply:
Thanks, good point. Let me remove this commit from this pull request, since it's unrelated to the SHA fix and add test cases for ensureStatefulSetPods()